File: solutions/elixir/lasagna/2/lib/lasagna.ex
defmodule Lasagna do
  # TODO: define the 'expected_minutes_in_oven/0' function
  def expected_minutes_in_oven, do: 40
  # TODO: define the 'remaining_minutes_in_oven/1' function
  def remaining_minutes_in_oven(time_elapsed), do: expected_minutes_in_oven() - time_elapsed
  # TODO: define the 'preparation_time_in_minutes/1' function
  def preparation_time_in_minutes(layers), do: 2 * layers 
  # TODO: define the 'total_time_in_minutes/2' function
  def total_time_in_minutes(layers, time_elapsed), do: preparation_time_in_minutes(layers) + time_elapsed
  # TODO: define the 'alarm/0' function
  def alarm, do: "Ding!"
end

File: solutions/swift/lasagna/1/Sources/Lasagna/Lasagna.swift
let expectedMinutesInOven = 40

func remainingMinutesInOven(elapsedMinutes: Int) -> Int
{
  guard elapsedMinutes >= 0 else { return 0 }
  return expectedMinutesInOven - elapsedMinutes
}

func preparationTimeInMinutes(layers: Int) -> Int
{
  layers * 2
}

func totalTimeInMinutes(layers: Int, elapsedMinutes: Int) -> Int
{
  preparationTimeInMinutes(layers: layers) + elapsedMinutes
}